pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1060,6 +1031,72 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.ClientTLSS
 				tlsContext.CommonTlsContext.AlpnProtocols = util.ALPNInMesh
 			}
 		}
+
+	case networking.ClientTLSSettings_SIMPLE, networking.ClientTLSSettings_MUTUAL:
```

Comment:
why are these combined when you the immediate branch on the type?

Reply:
like why not just another `case`?

Reply:
For now I can split it but when looking for CredentialName set, it'll be common to these two modes and a bunch of logic will be similar. With that in mind I kept them in the same case but we can split it in here itself if it is important

Reply:
Split the two for now